arca/src/serde.rs

```diff
@@ -76,7 +79,8 @@ impl<R: Runtime> Serialize for Tuple<R> {
         S: serde::Serializer,
     {
         self.with_ref(|x| {
-            let mut s = serializer.serialize_tuple(self.len())?;
+            let mut s = serializer.serialize_tuple(x.len())?;
```

Comment:
I think we should be doing `serialize_seq` here, since we deserialize it as a seq. This is probably also why we needed the extra `serialize_element`; if we change it I think that'll go away.